frontend/src/lib/components/widgets/TextInput/TextInput.tsx

```diff
@@ -138,26 +142,19 @@ class TextInput extends React.PureComponent<Props, State> {
     // If the TextInput is *not* part of a form, we mark it dirty but don't
     // update its value in the WidgetMgr. This means that individual keypresses
     // won't trigger a script re-run.
-    if (!isInForm(this.props.element)) {
-      this.setState({ dirty: true, value })
-      return
-    }
-
-    // If TextInput *is* part of a form, we immediately update its widgetValue
-    // on text changes. The widgetValue won't be passed to the Python
-    // script until the form is submitted, so this won't cause the script
-    // to re-run. (This also means that we won't show the "Press Enter
-    // to Apply" prompt because the TextInput will never be "dirty").
-    this.setState({ dirty: false, value }, () =>
```

Comment:
Similar to the comment above, are we sure that removing this doesn't cause any unexpected issue. Is the value still somehow kept in sync with the widget manager state?

Reply:
In TextInput.tsx and TextArea.tsx, I removed the committing of value and made sure the value is always dirty when it's changed. The reason was because when you're in a form and we commit the value, then the value is not dirty and therefore the Press Enter to submit form text won't be shown.

Reply:
ok. I think the only part here I'm a bit worried about is that we restore the value from the widget manager in some situations: 

https://github.com/streamlit/streamlit/blob/bcaa14f76bc1e3e55bc558a75951420078248d1d/frontend/src/lib/components/widgets/TextInput/TextInput.tsx#L67

I think this mainly happens if the delta path changes for the widget (e.g. new elements got added above). Something to test would be this kind of setup:

```python
if st.button("Click me"):
   st.write("Hello World")

with st.form("my_form"):
   text = st.text_input("Type some text")
   submitted = st.form_submit_button("Submit")
   if submitted:
       st.write(text)
```

If you type in text into the text input and then click the `Click me` button (without submitting the text), is your typed-in text preserved in the input field or resetted?

Reply:
Oh... That's a great point and I will test this out later today!

Reply:
So I tested this out and it looks like it's working fine from my perspective. Technically, the text is not properly registered as dirty since I did not submit. Is that something to worry about? 

https://github.com/streamlit/streamlit/assets/16749069/aa4be37f-a178-4aec-8eb8-60e368bc520b

Reply:
Alright, I think the aspect with dirty is fine. Two additional scenarios to test 1) What if you don't change the value again after clicking Click me and directly submit? 2) Does it also work fine if you do the same steps a second time in the same session?

Reply:
>1) Does it also work fine if you do the same steps a second time in the same session?

Yep it should work. This is the test case you're asking about right? 

https://github.com/streamlit/streamlit/assets/16749069/908fae8f-f17b-4555-9a99-4704109b77f0

>2) What if you don't change the value again after clicking Click me and directly submit?

It should also work though I think is the below the right test case?

https://github.com/streamlit/streamlit/assets/16749069/ce53ecc2-01ee-48da-b22f-adab32c56571